RPIPs/RPIP-X.md

```diff
@@ -0,0 +1,308 @@
+
+## Specification
+
+### Definitions
+
+There are several terms used in this document that are worth defining explicitly to avoid misunderstanding and ambiguity.
+
+#### Principal
+
+An individual or entity that can be said to ‘own’ the protocol.
+
+#### Agent
+
+An individual or entity that provides some service to the principals, or holds power delegated by the principals.
+
+#### Power
+
+Power in this document is used to refer to the raw capability to do something. This refers to explicit permissions granted in the protocol’s smart contracts, administrative access to software platforms, and so on.
+
+A principal or agent has the power to perform action X when every limitation to their role in performing action X is self-applied, rather than being applied by another party.
+
+To give an example, the oDAO has the power to make changes to the Rocket Pool Protocol due to the permissions trustedNodes (in aggregate) hold in the protocol contracts, and because the limitations to the oDAO ratified by the pDAO are self-applied by the oDAO.
+
+Best expressed in the phrase: ‘Are we able to do X?’
+
+#### Authority
+
+Authority in this document is used to refer to the right to do something. This refers to the implicit and explicit decisions made by the community at large as to ‘who do we want to do X’.
+
+Authorities are a bit complex in that they often stem from a string of decision-making powers, some explicit and some explicit. This document seeks to document what clear authorities exist based on approved governance actions and their ramifications, but may not be able to capture the full intricacy of who has the "right" to take an action.
+
+A principal or agent has the authority to do X if the Rocket Pool community has explicitly or implicitly agreed that that principal or agent is allowed or supposed to be doing X.
+
+To give an example, the pDAO has the authority to make changes to the Rocket Pool Protocol due to the shared understanding of the community that registeredNodes are the ‘principals’ of the Rocket Pool protocol.
+
+Best expressed in the phrase: ‘Are we allowed to do X?’
+
+### Principals and Agents
+
+#### oDAO governance
+
+##### Description
+
+OracleDAO governance is defined as successful on-chain governance proposals voted on by oDAO members running oDAO nodes.
+
+##### Powers
+
+* The oDAO has the power to replace any contract (except RocketStorage) in the Rocket Pool protocol via the oDAO upgrade power.
+* The oDAO has the power to modify any non-protected parameter in the Rocket Pool protocol via the oDAO upgrade power.
+* The oDAO has the power to call core contract functions protected by the generic ‘onlyLatestNetworkContract’ required via the oDAO upgrade power. This has several implications, most notably:
```

Comment:
These 3 sub-bullets are a bit weird...
- The first one would require a contract upgrade with faked up contracts with appropriate names to fool the check in RocketVault `require(etherBalances[contractName] >= _amount, "Insufficient contract ETH balance");`. I don't believe they have immediate access unless I'm missing something.
- The second bullet would also require a contract upgrade for arbitrary withdrawals... or they could fake up a merkle tree, which is probably a lot easier and has no dependence on `onlyLatestNetworkContract`
- The third bullet is possible via `RocketNetworkPenalties.submitPenalty`. The critical check is `onlyTrustedNode`, not `onlyLatestNetworkContract`

Minor aside: why are we using front ticks ‘onlyLatestNetworkContract’ instead of backticks `onlyLatestNetworkContract`?